src/app/util/privilege-storage.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "privilege-storage.h"
+
+#include <zap-generated/access.h>
+
+#include <lib/support/CodeUtils.h>
+
+#include <cstdint>
+
+using chip::AttributeId;
+using chip::ClusterId;
+using chip::CommandId;
+using chip::EventId;
+
+namespace {
+
+static_assert(GENERATED_ACCESS_PRIVILEGE__VIEW == kMatterAccessPrivilegeView,
+              "Generated privilege value must match privilege value in code");
+static_assert(GENERATED_ACCESS_PRIVILEGE__OPERATE == kMatterAccessPrivilegeOperate,
+              "Generated privilege value must match privilege value in code");
+static_assert(GENERATED_ACCESS_PRIVILEGE__MANAGE == kMatterAccessPrivilegeManage,
+              "Generated privilege value must match privilege value in code");
+static_assert(GENERATED_ACCESS_PRIVILEGE__ADMINISTER == kMatterAccessPrivilegeAdminister,
+              "Generated privilege value must match privilege value in code");
```

Comment:
I guess I'm not quite following.  What I meant was that the generated code should include privilege-storage.h and then use the constants that defines.

In what sense would the generated headers be "buildable/useable on their own"?  They are defining input to the privilege-storage bits, and I think that's fine.

(I also think they should include the relevant cluster/attribute/etc id headers and use symbolic ids instead of raw numbers, by the way, which would largely obviate the need for a lot of the comments we're adding now to make the raw numbers meaningful...  but that part can be a followup, if we decide to do it.)

Reply:
OK I did the collapse of the constants by including the API header in the generated header. If you also want to later try using IDs instead of raw numbers, I can make an issue ticket for that, for when we are in cleanup phase. I think the comments still have value even so, because they show the relation/binding of the parallel arrays (e.g. can easily match lines). And the point of the parallel arrays is they are both typed and packed naturally, and also allow for good memory coherence (since they are searched each array at a time), otherwise I'd be fine with a struct, but I do think the parallel areas should be more efficient for us.

Reply:
I made issue #16578 for using symbolic IDs instead of raw numbers, in case we want to do that later.